apps/src/code-studio/initApp/project.js

```diff
@@ -1462,13 +1471,21 @@ var projects = (module.exports = {
       }
 
       if (JSON.stringify(currentSources) === JSON.stringify(newSources)) {
-        hasProjectChanged = false;
+        if (!projectChangedWhileSaveInProgress) {
+          hasProjectChanged = false;
+        }
+        projectSaveInProgress = false;
+        projectChangedWhileSaveInProgress = false;
         callCallback();
         return;
       }
 
       this.saveSourceAndHtml_(newSources, () => {
-        hasProjectChanged = false;
+        if (!projectChangedWhileSaveInProgress) {
+          hasProjectChanged = false;
+        }
+        projectSaveInProgress = false;
+        projectChangedWhileSaveInProgress = false;
         callCallback();
       });
```

Comment:
Rather than adding the new variable, would moving the reset of hasProjectChanged to here solve the problem?
i.e. here we could do this: `hasProjectChanged = JSON.stringify(currentSources) !== JSON.stringify(newSources)`

Reply:
My initial reasoning for doing this was I thought we didn't want to reset the variable if the save failed, but I think we might call the callback even if the save fails (will need to double check that). I think there would still be a short period of time between getting the latest code and setting the variable in the case you suggested, but it might be short enough to not be an issue?

Reply:
looks like this callback doesn't get called on a failed save, but we do reset currentSources on failed save so my logic here may not be necessary